prody/proteins/interactions.py

```diff
@@ -160,6 +160,53 @@ def get_energy(pair, source):
     import numpy as np
     import importlib.resources as pkg_resources    
     
+    aa_correction = {
+        # Histidine (His)
+        'HSD': 'HIS',   # Protonated at ND1 (HID in AMBER)
+        'HSE': 'HIS',   # Protonated at NE2 (HIE in AMBER)
+        'HSP': 'HIS',   # Doubly protonated (HIP in AMBER)
+        'HID': 'HIS',   # AMBER name, protonated at ND1
+        'HIE': 'HIS',   # AMBER name, protonated at NE2
+        'HIP': 'HIS',   # AMBER name, doubly protonated
+        'HISD': 'HIS',  # GROMACS: protonated at ND1
+        'HISE': 'HIS',  # GROMACS: protonated at NE2
+        'HISP': 'HIS',  # GROMACS: doubly protonated
+
+        # Cysteine (Cys)
+        'CYX': 'CYS',   # Cystine (disulfide bridge)
+        'CYM': 'CYS',   # Deprotonated cysteine, anion (GROMACS)
+
+        # Aspartic acid (Asp)
+        'ASH': 'ASP',   # Deprotonated Asp (GROMACS: ASH, AMBER: AS4)
```

Comment:
I think this should say protonated not deprotonated, but please check. ASP and GLU generally have no protons on the side chain carboxylate

Reply:
Also you have ASH and GLH below so should these be AS4 and GL4?

Reply:
For AMBER, those residues are well documented:
https://ambermd.org/tutorials/basic/tutorial7/index.php
or https://emleddin.github.io/comp-chem-website/AMBERguide-protonation.html

NAMD I know well, so I can easily add those, but I cannot find such a list with non-standard names for Gromacs. So it is hard to double-check those. James, you using Gromacs, right? If you could confirm those it would be great.

Reply:
I don’t usually see non standard residues from gromacs but I can check it

The ions are pretty non standard in gromacs with the latest charmm36 force field though, e.g. CAL for calcium (to not match CA in protein) and CU2P for copper (II) positive. There was less of this with earlier versions

Reply:
> I don’t usually see non standard residues from gromacs but I can check it
> 
> The ions are pretty non standard in gromacs with the latest charmm36 force field though, e.g. CAL for calcium (to not match CA in protein) and CU2P for copper (II) positive. There was less of this with earlier versions

I just noticed this your previous message with https://manual.gromacs.org/documentation/current/reference-manual/topologies/pdb2gmx-input-files.html#residue-to-building-block-database

We can include those and that is it.

Reply:
My point with that link was to exactly not include those because they’re only used internally, but I can check it

Reply:
Ok, but it looks like some of them are the same in AMBER (see links above).
In NAMD those names are in PDB so we will have them after uploading PDB and trajectory into ProDy. I don't know how it is in Gromacs. I will trust you in this case.

Reply:
It looks like gromacs outputs 4 character names like HISE and HISD so they are usually parsed as HIS but the information is also there if people want it.

Reply:
Should I change something in the code, then? Do you have any other comments on the code, or can it be accepted in the current version?

Reply:
I think we should keep the gromacs things like we have then in case people parse long resnames, which I think there was an option for. 

I’ll have another look but I guess we can merge it

Reply:
Ok. I am making some new improvements to the old InSty code to analyze chains more conveniently, but this should be separate pull request because it may be more radical and should be better tested.

Reply:
ok, sounds good. you can make another branch for that then